tests/unit/test_security/test_session_auth.py

```diff
@@ -80,7 +86,13 @@ def test_session_auth_openapi(session_backend_config_memory: "ServerSideSessionC
         retrieve_user_handler=retrieve_user_handler,
         session_backend_config=session_backend_config_memory,
     )
-    app = Litestar(on_app_init=[session_auth.on_app_init])
+    from litestar.security.session_auth.plugin import SessionPlugin
+
+    app = Litestar(
+        # middleware=[SessionMiddleware(ServerSideSessionBackend(session_backend_config_memory)),
+        #             SessionAuthMiddleware(session_auth)],
+        plugins=[SessionPlugin(session_auth)],
```

Comment:
kept this now I remember to "showcase" the plugin is the same essentially as stacking the middlewares, now should you change their order and you're f**,